bin/testObservability/crashReporter/index.js

```diff
@@ -0,0 +1,146 @@
+  let local_path = "";
+  if(process.env["browserStackCwd"]){
+   local_path = path.join(process.env["browserStackCwd"], 'node_modules', module);
+  } else if(internal) {
+    local_path = path.join(process.cwd(), 'node_modules', 'browserstack-cypress-cli', 'node_modules', module);
+  } else {
+    local_path = path.join(process.cwd(), 'node_modules', module);
+  }
+  if(!fs.existsSync(local_path)) {
+    let global_path;
+    if(['jest-runner', 'jest-runtime'].includes(module))
+      global_path = path.join(GLOBAL_MODULE_PATH, 'jest', 'node_modules', module);
+    else
+      global_path = path.join(GLOBAL_MODULE_PATH, module);
+    if(!fs.existsSync(global_path)) {
+      throw new Error(`${module} doesn't exist.`);
+    }
+    return require(global_path);
+  }
+  return require(local_path);
+}
+
+getPackageVersion = (package_) => {
+  if(packages[package_]) return packages[package_];
+  return packages[package_] = this.requireModule(`${package_}/package.json`).version;
+}
+
+getAgentVersion = () => {
+  let _path = path.join(__dirname, '../../../package.json');
+  if(fs.existsSync(_path))
+    return require(_path).version;
+}
+
+class CrashReporter {
+  static instance;
+
+  constructor() {
+  }
+
+  static getInstance() {
+    if (!CrashReporter.instance) {
+      CrashReporter.instance = new CrashReporter();
+    }
+    return CrashReporter.instance;
+  }
+
+  setCredentialsForCrashReportUpload(credentialsStr) {
+    /* User credentials used for reporting crashes */
+    this.credentialsForCrashReportUpload = JSON.parse(credentialsStr);
+  }
+
+  setConfigDetails(credentialsStr, browserstackConfigFile, cypressConfigFile) {
+    /* User test config for build run */
+    this.userConfigForReporting = {
+      framework: 'Cypress',
+      browserstackConfigFile: browserstackConfigFile,
+      cypressConfigFile: cypressConfigFile
+    };
+    this.setCredentialsForCrashReportUpload(credentialsStr);
+  }
+
+  uploadCrashReport(exception, stacktrace) {
+    try {
+      if (!this.credentialsForCrashReportUpload.username || !this.credentialsForCrashReportUpload.password) {
+        return debug('[Crash_Report_Upload] Failed to parse user credentials while reporting crash')
+      }
+  
+      const data = {
+          hashed_id: process.env.BS_TESTOPS_BUILD_HASHED_ID,
+          observability_version: {
+              frameworkName: 'Cypress',
+              frameworkVersion: getPackageVersion('cypress'),
+              sdkVersion: getAgentVersion()
+          },
+          exception: {
+              error: exception.toString(),
+              stackTrace: stacktrace
+          },
+          config: this.userConfigForReporting
+      }
+  
+      const options = {
+        auth: {
+          ...this.credentialsForCrashReportUpload
+        },
+        headers: {
+          'Content-Type': 'application/json',
+          'X-BSTACK-TESTOPS': 'true'
+        },
+        method: 'POST',
+        url: `${API_URL}/api/v1/analytics`,
```

Comment:
lets more to config.js, makes easy to track all urls, in future to reuse.

Reply:
Discussed f2f, keeping testObservability config separate.